core/src/main/java/org/mule/util/ArrayUtils.java

```diff
@@ -138,5 +141,57 @@ else if (componentType.equals(Double.TYPE))
         result.retainAll(Arrays.asList(b));
         return result.toArray(new String[result.size()]);
     }
-    
+
+    public static int getLength(Object array)
+    {
+        if (array == null)
+        {
+            return 0;
+        }
+
+        checkArgument(array.getClass().isArray(), String.format("Object of type %s is not an array", array.getClass().getName()));
+
+        Class<?> componentType = array.getClass().getComponentType();
```

Comment:
I have no idea if it's necessary but just my 2 cents, lists and maps could be measured too: http://grepcode.com/file/repo1.maven.org/maven2/com.rabbitmq/amqp-client/2.3.0/com/rabbitmq/client/impl/Frame.java#284